test/geometry/liegroup/test_so3.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+import torch
+
+from kornia.geometry.liegroup.so3 import So3
+from kornia.geometry.quaternion import Quaternion
+
+from kornia.testing import assert_close
+
+
+class TestSo3:
+    @pytest.mark.parametrize("batch_size", (1, 2, 5))
+    def test_exp_log(self, device, dtype, batch_size):
+        q1 = Quaternion.random(batch_size)
+        q1 = q1.to(device, dtype)
+        s = So3(q1)
+        a = torch.rand(batch_size, 3, device=device, dtype=dtype)
```

Comment:
I would to give a try to test also as input numpy/numba and see how far we can go here.

also try to test the other members functions separately